src/blocks/services/service/edit.js

```diff
@@ -12,151 +13,115 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component, Fragment } from '@wordpress/element';
+import { Fragment, useEffect } from '@wordpress/element';
 import {
 	__experimentalImageURLInputUI as ImageURLInputUI,
 	BlockControls,
 	InnerBlocks,
 	MediaPlaceholder,
 } from '@wordpress/block-editor';
-import {
-	DropZone,
-	Button,
-	Spinner,
-	ButtonGroup,
-} from '@wordpress/components';
-import { dispatch, select } from '@wordpress/data';
+import { DropZone, Button, Spinner, ButtonGroup } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { mediaUpload } from '@wordpress/editor';
 import { isBlobURL } from '@wordpress/blob';
 import { closeSmall } from '@wordpress/icons';
+import { createBlock } from '@wordpress/blocks';
 
-/**
- * Constants
- */
-const ALLOWED_BLOCKS = [ 'core/heading', 'core/button', 'core/paragraph' ];
+const Edit = ( props ) => {
+	const {	attributes, clientId, setAttributes } = props;
+
+	const { getBlocksByClientId, getMedia, isSelected } = useSelect( ( select ) => {
+		const {	getBlockHierarchyRootClientId, getSelectedBlockClientId } = select( 'core/block-editor' );
+
+		// Get clientID of the parent block.
+		const rootClientId = getBlockHierarchyRootClientId( clientId );
+		const selectedRootClientId = getBlockHierarchyRootClientId( getSelectedBlockClientId() );
 
-class Edit extends Component {
-	constructor() {
-		super( ...arguments );
+		return {
+			getBlocksByClientId: select( 'core/block-editor' ).getBlocksByClientId,
+			getMedia: select( 'core' ).getMedia,
+			isSelected: isSelected || rootClientId === selectedRootClientId,
```

Comment:
I'm not seeing `isSelected` defined anywhere and beware that `isSelected` is passed into the props so you would be overwriting it here. I would suggest just changing this to `props.isSelected` rather than deconstructing it from the `props` var just for visibility into it being overridden here.